lib/src/migrators/module/scope.dart

```diff
@@ -58,83 +65,90 @@ class Scope {
       parent != null &&
       (functions.containsKey(name) || parent.isLocalFunction(name));
 
-  /// Constructs and throws a MigrationException for a reference to a member
-  /// from a nested import.
-  void _errorNestedImport(String type, Uri importUrl, FileSpan reference) {
-    throw MigrationException(
-        "Can't reference $type from ${p.prettyUri(importUrl)}, "
-        "as load-css() does not load ${type}s.",
-        span: reference);
-  }
-
-  /// Checks that [node] does not reference a variable from a nested import,
-  /// throwing a MigrationException if it does.
-  void checkNestedImportVariable(VariableExpression node) {
-    if (variables.containsKey(node.name)) {
-      var url = variables[node.name].span.sourceUrl;
-      if (nestedImports.contains(url)) {
-        _errorNestedImport('variable', url, node.span);
+  /// Checks whether a reference to a variable, mixin, or function is invalid,
+  /// throwing a MigrationException if it is.
+  ///
+  /// [findDeclaration] should return the declaration in the current scope for
+  /// the reference if it exists, or null if it does not.
+  void _checkUnreferencable(String type, FileSpan reference,
+      SassNode Function(Scope) findDeclaration) {
+    var declaration = findDeclaration(this);
+    if (declaration != null) {
+      if (unreferencableMembers.containsKey(declaration)) {
+        var uri = p.prettyUri(declaration.span.sourceUrl);
+        var unrefType = unreferencableMembers[declaration];
+        if (unrefType == UnreferencableType.localFromImporter) {
+          throw MigrationException(
+              "This stylesheet was loaded by a nested import in $uri. The "
+              "module system only supports loading nested CSS using the "
+              "load-css() mixin, which doesn't allow access to local ${type}s "
+              "from the outer stylesheet.",
+              span: reference);
+        } else if (unrefType == UnreferencableType.globalFromNestedImport) {
+          throw MigrationException(
+              "This $type was loaded from a nested import of $uri. The module "
+              "system only supports loading nested CSS using the load-css() "
+              "mixin, which doesn't load ${type}s.",
+              span: reference);
+        }
       }
     } else {
-      parent?.checkNestedImportVariable(node);
+      parent?._checkUnreferencable(type, reference, findDeclaration);
```

Comment:
Consider putting this clause first and returning early to avoid extra indentation:

```dart
if (declaration == null) {
  parent?._checkUnreferencable(type, reference, findDeclaration);
  return;
}
if (!unreferencableMembers.containsKey(declaration)) return;

// ...
```